tests/Connection/CachedQueryTest.php

```diff
@@ -83,4 +94,13 @@ private function createConnection(int $expectedQueryCount, array $columnNames, a
 
         return new Connection([], $driver);
     }
+
+    /** @return array<non-empty-string, list<callable():CacheItemPoolInterface>> */
+    public static function providePsrCacheImplementations(): array
+    {
+        return [
+            'serialized' => [static fn () => new ArrayAdapter(0, true)],
+            'by-reference' => [static fn () => new ArrayAdapter(0, false)],
```

Comment:
Why do you return closures instead of the actual adapters?

Reply:
Because `testCachedQuery` needs one Adapter while `testCachedQueryWithChangedImplementationIsExecutedTwice` needs two separate instances.